modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -10,20 +11,30 @@ const preventSummaryToggle = (editor: Editor): void => {
   });
 };
 
-// Forces the details element to always be open within the editor
 const filterDetails = (editor: Editor): void => {
   editor.parser.addNodeFilter('details', (elms) => {
+    const initialStateOption = Options.getDetailsInitialState(editor);
     Arr.each(elms, (details) => {
-      details.attr('data-mce-open', details.attr('open'));
-      details.attr('open', 'open');
+      if (initialStateOption === 'expanded') {
+        details.attr('open', 'open');
+      } else if (initialStateOption === 'collapsed') {
+        details.attr('open', null);
+      } else if (initialStateOption === 'inherited') {
```

Comment:
Could we just skip reassigning the same value?

Reply:
Yeah, it is totally redundant :)